tests/RobotFramework/tests/tedge_agent/workflows/run_workflow_on_child_device.robot

```diff
@@ -0,0 +1,66 @@
+*** Settings ***
+Resource            ../../../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs    name=${PARENT_SN}
+
+Test Tags           theme:tedge_agent
+
+
+*** Variables ***
+${PARENT_IP}    ${EMPTY}
+${PARENT_SN}    ${EMPTY}
+${CHILD_SN}     ${EMPTY}
+
+
+*** Test Cases ***
+Run workflow on child device
+    Set Device Context    ${PARENT_SN}
+    Execute Command
+    ...    tedge mqtt pub --retain te/device/${CHILD_SN}///cmd/user-command '{}'
```

Comment:
This message should be published by the child's tedge-agent.

Reply:
Oh, why? I thought any devices can publish a local command (and possible technically as the test is)

Reply:
I changed the order so that it is published by child device (inside of Child context)

Reply:
I have not been clear enough. It's useless to publish this capability message: this is done by the agent when the user-command workflow is added. If not this is a bug. So, I would do the reverse : i.e. check that the child agent actually publishes its capability to handle user-commands.

Reply:
Changed it to check the capability 👍

Reply:
Perfect. Thanks